### PR TITLE
Fix team selection

### LINKED ISSUE #121: Fix dropdown of teams for organizers
Dropdown on judging tab for organizers shows undefinedALL for all entries

### PR DESCRIPTION
**Pull request by abi-kothapalli**
Fixes N-W56 where all listings in team selection dropdown for organizers showed "UndefinedALL".

Previous code attempted to access non-existing fields of the team data. Call to API to retrieve teams also returned different data types for organizers vs judges, and trying to cast from one data type to another to show the team dropdown created issues for the organizer view. I simply standardized the naming conventions for the team data so that casting between the data types for organizers and judges should now be fine. 

Closes #121

### CODE REVIEW

**Comment by aadibajpai** [comment]
@abi-kothapalli can you pull in code from the current main so then it is easier to test your pr out by using the dev credentials?

**Comment by abi-kothapalli** [comment]
**Previous issues:**
- When an organizer went to the judging page, they would by default see the "No team selected" message and no data would be shown.
- Then, if the organizer selected a team, they would see the judging results for _all_ teams (not just the team they selected), regardless of what team they chose.

**New changes/fixes:**
- Judging page for organizers now displays table with judging results for all teams by default (before selecting a team from the dropdown). 
- Then, if a team is selected, the table will only display the judging results for that team.
- Export to CSV button should also work as expected (if a team is selected, only that team's data will be in the CSV) once #135 is merged.

**Comment by aadibajpai** [comment]
> 

looks good, last thing left is that if I select a team to narrow down the submissions, there is no way to go back to seeing all submissions

**Comment by Ace314159** [comment]
We'll merge this after #133  is merged. You'll have to rebase on his changes. Sorry about that.

File: types/client.ts [hunk 1]
```diff
@@ -29,8 +29,8 @@ export interface JudgingFormFields {
...
-	teamID: string;
-	teamName: string;
-	isMine: boolean;
-	haveJudged: boolean;
-}
\ No newline at end of file
+	_id: string;
+	name: string;
+	isMine?: boolean;
+	haveJudged?: boolean;
```
Comment (Ace314159) [inline types/client.ts:35]: Why are these optional?

File: components/allScores.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ export interface AllScoresProps {
 	scoreData: ScoreData[];
 	teamData: TeamData[];
 	userData: UserData[];
+	teamId?: string | null;
```
Comment (Ace314159) [inline components/allScores.tsx:11]: Can you not just use the id from `teamData`?

**Comment by abi-kothapalli** [comment]
If you select a team, you can now click the little 'x' on the dropdown to go back to seeing all submissions.

**Comment by aadibajpai** [comment]
hype, just need the conflicts resolved finally. @nkalupahana would you like to take a look and pick and choose

**Pull request merged**